docs/src/main/sphinx/develop/table-functions.rst

```diff
@@ -0,0 +1,133 @@
+engine during the Analysis phase of query processing. The ``analyze()`` method
+is also the place to perform custom checks on the arguments:
+
+.. code-block:: java
+
+    @Override
+    public Analysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        long columnCount = (long) ((ScalarArgument) arguments.get("column_count")).getValue();
+        long rowCount = (long) ((ScalarArgument) arguments.get("row_count")).getValue();
+
+        // custom validation of arguments
+        if (columnCount < 1 || columnCount > 3) {
+             throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "column_count must be in range [1, 3]");
+        }
+
+        if (rowCount < 1) {
+            throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "row_count must be positive");
+        }
+
+        // determine the returned row type
+        List<Descriptor.Field> fields = List.of("col_a", "col_b", "col_c").subList(0, (int) columnCount).stream()
+                .map(name -> new Descriptor.Field(name, Optional.of(BIGINT)))
+                .collect(toList());
+
+        Descriptor returnedType = new Descriptor(fields);
+
+        return new Analysis(
+                Optional.of(returnedType),
+                EMPTY_MAPPING,
+                new ConnectorTableFunctionHandle() {});
+    }
+
+The ``analyze()`` method returns an ``Analysis`` object, which comprises all
+the information required by the engine to analyze, plan and execute the table
+function invocation:
+
+- the returned row type, specified as an optional ``Descriptor``. It should be
+  passed if and only if the table function is declared with the
+  ``GENERIC_TABLE`` returned type.
+- dependencies between descriptor arguments and table arguments. Currently
+  descriptor and table arguments are not supported, so it should be
+  ``EMPTY_MAPPING``.
+- any information gathered during analysis that might be useful during planning
+  or execution, in the form of a ``ConnectorTableFunctionHandle``.
+  ``ConnectorTableFunctionHandle`` is a marker interface intended to carry
+  information throughout subsequent phases of query processing in a manner that
+  is opaque to the engine.
+
+Table function execution
+------------------------
+
+Table functions are executed as pushdown to the connector. The connector that
+provides a table function should implement the ``applyTableFunction()`` method.
+This method is called during the Optimization phase of query processing. It
+returns a ``ConnectorTableHandle`` and a list of ``ColumnHandle`` s
+representing the table function result. The table function invocation is then
+replaced with a ``TableScanNode``.
+
+Access control
+--------------
+
+The access control for table functions can be provided both on system and
+connector level. It is based on the fully qualified table function name,
+which consists of the catalog name, the schema name, and the function name.
```

Comment:
in the syntax of ?

catalog.schema.function

I assume.. but spell that out..